docs/users-permissions.md

```diff
@@ -0,0 +1,60 @@
+- users with `superuser` status,
+- users who belong to the `admin` group.
+
+### Object-level permissions
+We need a way to attach specific users to existing `Hostingprovider` objects in the system to ensure that only authorized users are able to make updates to the data they submitted. This is enforced by object-level permission system, which implementation relies on the library [`django-guardian`](https://django-guardian.readthedocs.io/en/stable/).
+
+Currently we have the following permissions defined:
+- `manage_provider`: ties a specific user to a specific `Hostingprovider` object, 
+- `manage_datacenter`: ties a specific user to a specific `Datacenter` object.
+
+The mechanism provided by `django-guardian` allows us to add arbitrary number of permissions. For example:
+- a single user can manage multiple providers/datacenters,
+- a single provider/datacenter can be managed by multiple users.
+
+Various parts of the system (Provider Portal, API views and Admin views) will inspect the object-level permission to determine whether the authenticated users is authorized to view/edit given object. As an example: if user `alice` wants to upload new evidence to the existing hosting provider `Green Provider`, the system will check the rule: does the permission `manage_provider` exist for user `alice` and object `Green Provider`? The same mechanism is used in places where we list all providers attached to a given user, e.g. Provider Portal (retrieve all objects for which `alice` has the permission `manage_provider`). And same for places where we list all users attached to a given provider, e.g. Django Admin page for `Green Provider` will retrieve all users who have the permission `manage_provider` for the object `Green Provider`.
+
+The mechanism provided by `django-guardian` can be extended to groups, too. Currently we use it to assign a global permission (meaning: to all objects in the system) for the `admin` group. As a consequence, all `admin` group members have access to all hosting providers and all datacenters. Given that there are thousands of providers in our Green Web Database, listing all of them in the Provider Portal for `admin` group members could be cumbersome (potential performance issues) and not useful (the rendered list will take ages to scroll through). That's why wherever we list all objects for a given user (see `alice` example above), we ignore global permissions and those that are connected to group membership, leaving only explicit user-object permissions. To illustrate that let's extend the example above:
+- user `alice` from the example above belongs to the `admin` group, 
+- there are 5000 hosting provider objects in the database,
+- there exists only 1 explicit permission `manage_provider` for `alice`: that is to `Green Provider`.
+
+Even though `alice` as a member of the `admin` group can manage all 5000 providers, when she visits the Provider Portal she will only see `Green Provider` listed. 
```

Comment:
Small typo:

> Even though `alice` **is** a member

Given Alice is a name, it might make sense to have `Alice`, but this is less of a priority IMHO.